is_iam_enabled.go

```diff
@@ -0,0 +1,74 @@
+package alks
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+)
+
+type IsIamEnabledRequest struct {
+	Account string `json:"account,omitempty"`
+	Role    string `json:"role,omitempty"`
+	RoleArn string `json:"roleArn,omitempty"`
+}
+
+// IsIamEnabledResponse is used to represent a role that's IAM active or not.
+type IsIamEnabledResponse struct {
+	BaseResponse
+	Account    string `json:"account,omitempty"`
+	Role       string `json:"role,omitempty"`
+	RoleArn    string `json:"roleArn"`
+	IamEnabled bool   `json:"iamEnabled"`
+}
+
+// IsIamEnabled will check if a MI, AccountDetails, or STS assumed role is IAM active or not.
+func (c *Client) IsIamEnabled(roleArn string) (*IsIamEnabledResponse, error) {
+
+	if len(roleArn) > 1 {
+		log.Printf("[INFO] Is IAM enabled for MI: %s", roleArn)
+	} else {
+		log.Printf("[INFO] Is IAM enabled for: %s", c.AccountDetails.Account)
+		log.Printf("[INFO] Is IAM enabled for: %s", c.AccountDetails.Role)
+	}
+
+	iam := IsIamEnabledRequest{
+		c.AccountDetails.Account,
+		c.AccountDetails.Role,
+		roleArn,
+	}
+
+	body, err := json.Marshal(struct {
+		IsIamEnabledRequest
+	}{iam})
```

Comment:
We don't need this extra boilerplate/allocation here.  `json.Marshal` will take our request type directly.  Someday we'll get around to refactoring all the other places it's called this way, but for now let's just start doing it the proper way going forward.
```suggestion
	body, err := json.Marshal(iam)
```